crates/red_knot_python_semantic/src/semantic_index.rs

```diff
@@ -990,4 +990,28 @@ def x():
             vec!["bar", "foo", "Test", "<module>"]
         );
     }
+
+    #[test]
+    fn match_stmt_symbols() {
+        let TestCase { db, file } = test_case(
+            "
+match subject:
+    case a: ...
+    case [b, c, *d]: ...
+    case e as f: ...
+    case {'x': g, **h}: ...
+    case Foo(i, z=j): ...
+    case k | l: ...
+    case _: ...
```

Comment:
At runtime this pattern does actually bind the name `_`: 

```
>>> match 3:
...     case _: pass
...
>>> _
3
```

So perhaps we should as well?

Reply:
Wait what? I tried this locally and it gives me a `NameError` on 3.10 to 3.13:
```py
match 1:
    case _:
        print(_)
```

Running it from the command-line:
```console
✘ python3.13 type_inference/stmt_match.py
Traceback (most recent call last):
  File "/Users/dhruv/playground/ruff/type_inference/stmt_match.py", line 5, in <module>
    print(_)
          ^
NameError: name '_' is not defined
```

Maybe this is specific to the REPL?

Reply:
The REPL always binds the `_` symbol to the result of whatever expression was last evaluated, so this could indeed be something REPL-specific

Reply:
Ah, good call! Totally forgot about that REPL feature. That's a funny combination of separate special-casing around underscores, it really looked like the match statement was binding 😆